xarray/core/dataset.py

```diff
@@ -306,7 +309,7 @@ def load(self):
         """
         # access .data to coerce everything to numpy or dask arrays
         all_data = dict((k, v.data) for k, v in self.variables.items())
-        lazy_data = dict((k, v) for k, v in all_data.items()
+        lazy_data = OrderedDict((k, v) for k, v in all_data.items()
```

Comment:
I don't think this matters. Variables are already assigned in the proper order in `self.variables`, and we never return this object to the user.